src/renderer/store/handlers/waggle.ts

```diff
@@ -0,0 +1,50 @@
+
+import { createAction, handleActions } from 'redux-actions'
+import { actionTypes } from '../../../shared/static'
+import { produce, immerable } from 'immer'
+
+import { ActionsType, PayloadType } from './types'
+
+export class Waggle {
+  isWaggleConnected: boolean
+
+  constructor(values?: Partial<Waggle>) {
+    Object.assign(this, values)
+    this[immerable] = true
+  }
+}
+
+export type WaggleStore = Waggle
+
+export const initialState: Waggle = {
+  isWaggleConnected: false
+}
+
+const setIsWaggleConnected = createAction<boolean>(actionTypes.SET_IS_WAGGLE_CONNECTED)
+
+export const actions = {
+  setIsWaggleConnected
+}
+
+export type WaggleActions = ActionsType<typeof actions>
+
+export const epics = {}
+
+export const reducer = handleActions<WaggleStore, PayloadType<WaggleActions>>(
```

Comment:
same reducer created with @reduxjs/toolkit looks like this:

```
import { createAction, createReducer, PayloadAction } from '@reduxjs/toolkit'

export const reducer = createReducer<WaggleStore>(initialState, (builder) => {
  builder.addCase(setIsWaggleConnected, (state, action: WaggleActions['setIsWaggleConnected']) => {
            state.isWaggleConnected = action.payload    
  })
})

```

I personally find it easier for the eye. Reduxjs toolkit has immer build into it, so no need for explicit produce call

Reply:
I would prefer we start using this lib for new reducers and at some point refactor the old ones too